bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/HistoryStore.java

```diff
@@ -0,0 +1,105 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Vector Informatik GmbH and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Vector Informatik GmbH - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.ui.internal.findandreplace;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.eclipse.jface.dialogs.IDialogSettings;
+
+/**
+ * Store for search histories or replace histories inside of a dialog. Stores
+ * the nodes using the DialogSettings mechanism.
+ */
+public class HistoryStore {
+	IDialogSettings settingsManager;
+	private int historySize;
+	List<String> history;
+	String sectionName;
+
+	/**
+	 * @param settingsManager manager for DialogSettings
+	 * @param sectionName     the name of the section in the DialogSettings
+	 *                        containing the history
+	 * @param historySize     how many entries to keep in the history
+	 */
+	public HistoryStore(IDialogSettings settingsManager, String sectionName, int historySize) {
+		this.settingsManager = settingsManager;
+		this.historySize = historySize;
+		loadSection(sectionName);
+	}
+
+	public Iterable<String> get() {
+		return history;
+	}
+
+	public String get(int index) {
+		return history.get(index);
+	}
+
+
+	public void add(String historyItem) {
+		if (sectionName == null) {
+			throw new IllegalStateException("No section loaded"); //$NON-NLS-1$
+		}
+		if (historyItem != null && !historyItem.isEmpty()) {
+			history.add(0, historyItem);
+		}
+
+		writeHistory();
+	}
+
+	public void remove(String historyItem) {
+		history.remove(history.indexOf(historyItem));
+	}
+
+	public boolean isEmpty() {
+		return history.isEmpty();
+	}
+
+	private void loadSection(String newSectionName) {
+		this.sectionName = newSectionName;
+
+		history = new ArrayList<>(Arrays.asList(settingsManager.getArray(newSectionName)));
```

Comment:
This breaks as `settingsManager.getArray` can return `null`. This is also the reason why the reflective instantiation in the old `FindReplaceDialogTest` breaks. When fixing this, at least two of the existing `FindReplaceDialogTests` will work again.